src/auditlog/models.py

```diff
@@ -232,6 +237,73 @@ def changes_str(self, colon=': ', arrow=smart_text(' \u2192 '), separator='; '):
+        for field_name, values in iteritems(self.changes_dict):
+            # try to get the field attribute on the model
+            try:
+                field = model._meta.get_field(field_name)
+            except FieldDoesNotExist:
+                # use the reverse field mapping to get the field
+                from auditlog.registry import auditlog
+                model_fields = auditlog.get_model_fields(model._meta.model)
+                reversed_field = model_fields['reverse_mapping_fields'].get(field_name)
+                field = model._meta.get_field(reversed_field)
+
+            values_display = []
+            # handle choices fields and Postgres ArrayField to get human readable version
+            if field.choices or hasattr(field, 'base_field') and getattr(field.base_field, 'choices', False):
+                choices_dict = dict(field.choices or field.base_field.choices)
+                for value in values:
+                    try:
+                        value = ast.literal_eval(value)
+                        if type(value) is [].__class__:
+                            values_display.append(', '.join([choices_dict.get(val, 'None') for val in value]))
+                        else:
+                            values_display.append(choices_dict.get(value, 'None'))
+                    except ValueError:
+                        values_display.append(choices_dict.get(value, 'None'))
+                    except:
+                        values_display.append(choices_dict.get(value, 'None'))
+            else:
+                for value in values:
+                    # handle case where field is a datetime type
+                    if "DateTime" in field.get_internal_type():
+                        try:
+                            value = parser.parse(value)
+                            value = value.strftime("%b %d, %Y %I:%M %p")
+                        except ValueError:
+                            pass
+                    # handle case where field is a date type
+                    elif "Date" in field.get_internal_type():
+                        try:
+                            value = parser.parse(value)
+                            value = value.strftime("%b %d, %Y")
+                        except ValueError:
+                            pass
+                    # handle case where field is a time type
+                    elif "Time" in field.get_internal_type():
+                        try:
+                            value = parser.parse(value)
+                            value = value.strftime("%I:%M %p")
+                        except ValueError:
+                            pass
+                    # check if length is longer than 140 and truncate with ellipsis
+                    if len(value) > 140:
+                        value = "{}...".format(value[:140])
+
+                    values_display.append(value)
+
+            changes_display_dict[field_name] = values_display
```

Comment:
Instead of the field name we might use the mapping here. Also, we might consider going with the verbose name if present. So in order of importance: mapping - verbose name - name (the last is only guaranteed to be present).

Reply:
Ok made this change, the order is mapping -> verbose name -> default verbose name. Django assigns a default verbose_name if none is provided so it is guaranteed to be always there.

>If this isn’t given, Django will use a munged version of the class name: CamelCase becomes camel case.
[ref](https://docs.djangoproject.com/en/1.11/ref/models/options/#verbose-name)